MahApps.Metro/Behaviours/DatePickerTextBoxBehavior.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace MahApps.Metro.Behaviours
+{
+    using System.Windows.Controls;
+    using System.Windows.Controls.Primitives;
+    using System.Windows.Interactivity;
+    using MahApps.Metro.Controls;
+
+    public class DatePickerTextBoxBehavior : Behavior<DatePickerTextBox>
+    {
+        protected override void OnAttached()
+        {
+            base.OnAttached();
+            AssociatedObject.TextChanged += OnTextChanged;
+        }
```

Comment:
@thoemmi thanks for pointing that out. An additional question is: `OnDetaching` is only called when attaching/detaching throug C# code not when attaching the behavior directly throug XAML, like it is done here. Do you think this might be a memory leak to or is this no problem?